zebra-network/src/peer/connection.rs

```diff
@@ -171,18 +192,30 @@ where
                         Either::Right((None, _)) => {
                             self.fail_with(PeerError::DeadClient);
                         }
-                        Either::Right((Some(req), _)) => self.handle_client_request(req).await,
+                        Either::Right((Some(req), _)) => {
+                            let span = req.span.clone();
+                            self.handle_client_request(req).instrument(span).await
+                        }
                     }
                 }
                 // We're awaiting a response to a client request,
                 // so wait on either a peer message, or on a request timeout.
-                State::AwaitingResponse { .. } => {
-                    trace!("awaiting response to client request");
+                State::AwaitingResponse { ref span, .. } => {
+                    // we have to get rid of the span reference so we can tamper with the state
+                    let span = {
+                        let span2 = span.clone();
+                        std::mem::drop(span);
+                        span2
+                    };
+                    trace!(parent: &span, "awaiting response to client request");
```

Comment:
I don't think this does anything other than clone the span. What are you trying to do here?

Reply:
I was trying to ensure that `span` (a reference to the state) didn't live long enough to interfere with modifications to the state.  But I guess that NLL is enough to do that, as long as the clone happens.